app/controllers/concerns/api/v2/parameter_deprecation.rb

```diff
@@ -0,0 +1,61 @@
+# This module adds a pre-action filter that will handle deprecated properties.
```

Comment:
That's even more complicated, why do you need to traverse a tree to rename a parameter? what's wrong with something along the lines of this, if you insist on generalizing this:
```ruby
   def deprecate_rename_param(old,new)
     if !params[old].nil?
       params[new] = params.delete(old)
       Foreman::Deprecation.api_deprecation_warning("#{old} was renamed to #{new}")
     end
   end
```